subprojects/language-native/src/integTest/groovy/org/gradle/language/swift/SwiftLibraryDependenciesIntegrationTest.groovy

```diff
@@ -16,29 +16,140 @@
 
 package org.gradle.language.swift
 
-import org.gradle.language.AbstractNativeProductionComponentDependenciesIntegrationTest
+import org.gradle.language.AbstractNativeLibraryDependenciesIntegrationTest
 import org.gradle.util.Requires
 import org.gradle.util.TestPrecondition
 
 @Requires(TestPrecondition.SWIFT_SUPPORT)
-class SwiftLibraryDependenciesIntegrationTest extends AbstractNativeProductionComponentDependenciesIntegrationTest {
-    @Override
-    protected void makeComponentWithLibrary() {
+class SwiftLibraryDependenciesIntegrationTest extends AbstractNativeLibraryDependenciesIntegrationTest {
+    def "implementation dependencies are visible to downstream consumers"() {
```

Comment:
I think we should rename this a bit.  We don't really _want_ this behavior, so maybe something like:

"can compile library against another library with implementation dependencies"

Reply:
Yep - good call.  Done.